### PR TITLE
Migrate token example

### PR DESCRIPTION
**Pull request by sisuresh**
Resolves https://github.com/stellar/soroban-token-contract/issues/107

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,7 @@ members = [
     "single_offer_router",
     "events",
     "timelock",
+    "token_contract"
```
Comment (leighmcculloch) [inline Cargo.toml:18]: I don't think the `_contract` suffix is necessary. We don't use it for others for their directory name.
```suggestion
    "token"
```

**Review by leighmcculloch — APPROVED** [review]
Thanks! One suggested change, otherwise lgtm.

I added @jonjove as a reviewer as well.

File: token_contract/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+testutils = ["soroban-sdk/testutils", "soroban-auth/testutils", "dep:ed25519-dalek"]
+
+[dependencies]
+ed25519-dalek = { version = "1.0.1", optional = true }
+num-bigint = { version = "0.4", optional = true }
+soroban-sdk = { version = "0.0.4" }
+soroban-auth = { version = "0.0.4" }
+
+[dev-dependencies]
+soroban-token-contract = { path = ".", features = ["export", "testutils"] }
```
Comment (leighmcculloch) [inline token_contract/Cargo.toml:22]: This self dependency needs removing. One of the builds is breaking showing that it is not compatible with rust-analyzer: 
```
[ERROR project_model::workspace] cyclic deps: soroban_token_contract(CrateId(98)) -> soroban_token_contract(CrateId(98)), alternative path: soroban_token_contract(CrateId(98))
```

File: Cargo.toml (hunk 1 above)
Comment (sisuresh) [reply Cargo.toml]: Done

File: token_contract/Cargo.toml (hunk 2 above)
Comment (sisuresh) [reply token_contract/Cargo.toml]: Removed

**Review by leighmcculloch — APPROVED** [review]

**Pull request merged**